lib/web/app/middleware.go

```diff
@@ -65,6 +65,11 @@ func (h *Handler) withAuth(handler handlerAuthFunc) http.HandlerFunc {
 // address of the proxy is set.
 func (h *Handler) redirectToLauncher(w http.ResponseWriter, r *http.Request, p launcherURLParams) error {
 	if h.c.WebPublicAddr == "" {
+		// The error below tends to be swallowed by the Web UI, so log a warning for
+		// admins as well.
+		h.log.Error("" +
+			"Application Service needs the Proxy's public_addr to be explicitly set, otherwise it won't work. " +
+			"Please refer to https://goteleport.com/docs/application-access/guides/connecting-apps/#start-authproxy-service.")
```

Comment:
How about something like:

```suggestion
			"Application Service requires public_addr to be set in the Teleport Proxy Service configuration. " +
			"Please contact your Teleport cluster administrator or refer to https://goteleport.com/docs/application-access/guides/connecting-apps/#start-authproxy-service.")
```

Otherwise it's a little unclear from the message whether it needs to be set in app service config or proxy config and the user may get confused.

Reply:
Done.